File: lab12/factors.cpp
#include <iostream>

using namespace std;

// finds all of n's factors
// not including 1 and itself
void factors (int n)
{
  int k = 2;

  while (k < n) {
    
    if (n % k == 0) {
      cout << k << endl;
    }
    k++;
  }
  
}

int main()
{

  int *a = 1;
  int b[2] = {2,2};
  int *c = &a;
  int* d = &b;

  cout << "a: " << a <<" - " << &a << endl;
  cout << "b: " << b <<" - " << &b[0] <<" - " << &b[1] << endl;
  cout << "c: " << c <<" - " << *c << endl;
  cout << "d: " << d <<" - " << *d << d[1] << endl;

  int n;
  cout << "Find factors of what number? ";
  cin >> n;
  cout << "Factors of n are: " << endl;
  factors (n);
  cout << endl;
}

File: lab6/Factors.java
import java.util.Scanner;

/**
   class contains static method to find all factors of n
   plus a test driver for the function
*/
public class Factors {


    /**
       finds all of n's factors
       not including 1 and itself
       pre: n >= 1
    */
    public static void factors (int n) {
	int k = 0;

	while (k < n) {
	    if (n % k == 0) {
		System.out.print(k + " ");
	    }
	}
  
    }


    public static void main(String[] args) {

	Scanner in = new Scanner(System.in);

	System.out.print("Find factors of what number? ");
	int n = in.nextInt();
	System.out.println("Factors of n are: ");

	factors (n);

	System.out.println();
    }

}
